go/vt/vtgate/semantics/info_schema_gen_test.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package semantics
+
+import (
+	"database/sql"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/vt/sqlparser"
+)
+
+func TestGenerateInfoSchemaMap(t *testing.T) {
+	t.Skip("run manually to re-create the content of the getInfoSchema function")
+	b := new(strings.Builder)
+
+	db, err := sql.Open("mysql", "root@tcp(127.0.0.1:3306)/test")
+	require.NoError(t, err)
+	defer db.Close()
+
+	for _, tbl := range informationSchemaTables {
+		b.WriteString("cols = map[string]query.Type{}\n")
+		result, err := db.Query(fmt.Sprintf("show columns from information_schema.`%s`", tbl))
+		require.NoError(t, err)
+		defer result.Close()
+		for result.Next() {
+			var r row
+			result.Scan(&r.Field, &r.Type, &r.Null, &r.Key, &r.Default, &r.Extra)
+			allString := re.FindStringSubmatch(r.Type)
+			var typ string
+			if allString == nil {
+				typ = r.Type
+			} else {
+				typ = allString[1]
+			}
+			unsigned := false
+			if idx := strings.Index(typ, "unsigned"); idx > 0 {
+				typ = typ[:idx-1]
+				unsigned = true
+			}
+			i2 := sqlparser.SQLTypeToQueryType(typ, unsigned)
+			if int(i2) == 0 {
+				t.Fatalf("%s %s", tbl, r.Field)
+			}
+			b.WriteString(fmt.Sprintf("cols[\"%s\"] = query.Type(%d)\n", r.Field, int(i2)))
+		}
+		b.WriteString(fmt.Sprintf("inf_schema[\"%s\"] = cols\n", tbl))
+	}
+
+	fmt.Println(b.String())
+}
+
+var informationSchemaTables = []string{
+	"ADMINISTRABLE_ROLE_AUTHORIZATIONS",
+	"APPLICABLE_ROLES",
+	"CHARACTER_SETS",
+	"CHECK_CONSTRAINTS",
+	"COLLATION_CHARACTER_SET_APPLICABILITY",
+	"COLLATIONS",
+	"COLUMN_PRIVILEGES",
+	"COLUMN_STATISTICS",
+	"COLUMNS",
+	"COLUMNS_EXTENSIONS",
+	"ENABLED_ROLES",
+	"ENGINES",
+	"EVENTS",
+	"FILES",
+	"INNODB_BUFFER_PAGE",
+	"INNODB_BUFFER_PAGE_LRU",
+	"INNODB_BUFFER_POOL_STATS",
+	"INNODB_CACHED_INDEXES",
+	"INNODB_CMP",
+	"INNODB_CMP_PER_INDEX",
+	"INNODB_CMP_PER_INDEX_RESET",
+	"INNODB_CMP_RESET",
+	"INNODB_CMPMEM",
+	"INNODB_CMPMEM_RESET",
+	"INNODB_COLUMNS",
+	"INNODB_DATAFILES",
+	"INNODB_FIELDS",
+	"INNODB_FOREIGN",
+	"INNODB_FOREIGN_COLS",
+	"INNODB_FT_BEING_DELETED",
+	"INNODB_FT_CONFIG",
+	"INNODB_FT_DEFAULT_STOPWORD",
+	"INNODB_FT_DELETED",
+	"INNODB_FT_INDEX_CACHE",
+	"INNODB_FT_INDEX_TABLE",
+	"INNODB_INDEXES",
+	"INNODB_METRICS",
+	"INNODB_SESSION_TEMP_TABLESPACES",
+	"INNODB_TABLES",
+	"INNODB_TABLESPACES",
```

Comment:
Does this mean we only support the 8.0 table names? E.g. `innodb_tablespaces` was called `innodb_sys_tablespaces` in MySQL 5.7, and I think there's others that got renamed as well.

Reply:
Should we support the "old" table names when the vtgate runs in mysql 5.7 mode?

Reply:
excellent point. yeah, we should support both

Reply:
added a commit for this now